esp-hal/src/timer/systimer.rs

```diff
@@ -146,90 +93,81 @@ impl<'d> SystemTimer<'d> {
+    #[cfg(not(esp32s2))]
+    /// Configures when this counter can run.
+    /// It can be configured to stall or continue running when CPU stalls
+    /// or enters on-chip-debugging mode.
     ///
-    /// This is a convenience method to create `'static` alarms of the same
-    /// type. You are encouraged to use [Alarm::new] over this very specific
-    /// helper.
-    pub fn split<MODE>(self) -> SysTimerAlarms<MODE, Blocking> {
-        static mut UNIT0: Option<AnyUnit<'static>> = None;
-        let unit0 = unsafe { &mut *addr_of_mut!(UNIT0) };
-
-        let unit0 = unit0.insert(self.unit0.into());
-        let unit = FrozenUnit::new(unit0);
-
-        SysTimerAlarms {
-            alarm0: Alarm::new(self.comparator0.into(), &unit),
-            alarm1: Alarm::new(self.comparator1.into(), &unit),
-            alarm2: Alarm::new(self.comparator2.into(), &unit),
-            #[cfg(not(esp32s2))]
-            unit1: self.unit1,
-        }
+    /// # Safety
+    ///
+    /// - Disabling an `Unit` whilst [`Alarm`]'s are using it will affect the
+    ///   [`Alarm`]'s operation.
+    /// - Disabling Unit0 will affect [`now`](crate::time::now).
+    pub unsafe fn configure_unit(unit: Unit, config: UnitConfig) {
+        unit.configure(config)
     }
 
-    /// Split the System Timer into three alarms.
+    /// Set the value of the counter immediately. If the unit is at work,
+    /// the counter will continue to count up from the new reloaded value.
     ///
-    /// This is a convenience method to create `'static` alarms of the same
-    /// type. You are encouraged to use [Alarm::new_async] over this very
-    /// specific helper.
-    pub fn split_async<MODE>(self) -> SysTimerAlarms<MODE, Async> {
-        static mut UNIT0: Option<AnyUnit<'static>> = None;
-        let unit0 = unsafe { &mut *addr_of_mut!(UNIT0) };
-
-        let unit0 = unit0.insert(self.unit0.into());
-        let unit = FrozenUnit::new(unit0);
-
-        SysTimerAlarms {
-            alarm0: Alarm::new_async(self.comparator0.into(), &unit),
-            alarm1: Alarm::new_async(self.comparator1.into(), &unit),
-            alarm2: Alarm::new_async(self.comparator2.into(), &unit),
-            #[cfg(not(esp32s2))]
-            unit1: self.unit1,
-        }
+    /// This can be used to load back the sleep time recorded by RTC timer
+    /// via software after Light-sleep
+    ///
+    /// # Safety
+    ///
+    /// - Modifying a unit's count whilst [`Alarm`]'s are using it may cause
+    ///   unexpected behaviour
+    /// - Any modification of the unit0 count will affect
+    ///   [`now`](crate::time::now).
+    pub unsafe fn set_unit_value(unit: Unit, value: u64) {
+        unit.set_count(value)
     }
 }
 
 /// A
 #[cfg_attr(esp32s2, doc = "64-bit")]
 #[cfg_attr(not(esp32s2), doc = "52-bit")]
 /// counter.
-pub trait Unit {
-    /// Returns the unit number.
-    fn channel(&self) -> u8;
+#[derive(Copy, Clone, Debug, Default, PartialEq, Eq)]
+pub enum Unit {
+    #[default]
```

Comment:
Considering that touching Unit0 can mess up the time keeping, do we really want to have a default?